### ISSUE TITLE
Support for swagger spec

### ISSUE THREAD

**Issue by gosuri** [adams-sarah/test2doc#1]
Awesome Library. The part I really like is ability to choose the APIs to document (other similar ones don't really provide that flexibility)

Will be even more awesome if you could tap into the [swagger](http://swagger.io) eco-system (auto client generators, api explorer UIs etc). 

Reference Spec - https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md

**Comment by yanpozka** [comment]
@gosuri thanks for open this issue, you did read my mind ;)

Hi @adams-sarah thanks so much for this repo, I really like it !! I'w like add swagger support to this tool, would you accept a possible PR with swagger support ?

**Comment by adams-sarah** [comment]
Sure! That would be amazing!
If you could just get please send me your design proposal once you get familiar with the code, before you write any new code. That way, neither of us wastes time!

Thanks!

**Comment by shauvik** [comment]
@yanpozka Thanks for volunteering. Swagger support would be awesome!
Do you have any updates on the PR?

**Comment by yanpozka** [comment]
@shauvik Yes swagger support will be great, sadly I've been busy in my work and I ended using this one: http://goswagger.io which I don't recommend it forces you to have structs for request and it's impossible use composition or golang inheritance
